launcher/launcher/main.go

```diff
@@ -188,6 +194,33 @@ func startLauncher(launchSpec spec.LaunchSpec, serialConsole *os.File) error {
 	}
 	defer tpm.Close()
 
+	// check DA info, don't crash if failed
+	daInfo, err := launcher.GetTPMDAInfo(tpm)
+	if err != nil {
+		log.Printf("failed to get DA Info: %v\n", err)
+	} else {
+		log.Printf("TPM DA params: %+v\n", daInfo)
+
+		if !daInfo.StartupClearOrderly {
+			log.Printf("TPM didn't start orderly, DA lockout counter incremented: %d/%d\n", daInfo.LockoutCounter, daInfo.MaxAuthFail)
+		}
+
+		// if DA params not as expected (first boot on the VM with a fresh TPM)
+		if daInfo.LockoutInterval != lockoutInterval || daInfo.MaxAuthFail != maxAuthFail || daInfo.LockoutRecovery != lockoutRecovery {
```

Comment:
Why can't we just update on each boot? Is there a downside?

Reply:
We could do that I just want avoid unnecessary TPM call..

Reply:
I just remove it for simplicity